packages/components/src/client/components/ArtPlayer.ts

```diff
@@ -0,0 +1,211 @@
+} from "../../shared/index.js";
+
+declare const ART_PLAYER: ComponentsArtPlayerOptions;
+
+const optionsDefault = {
+  width: "100%",
+  ratio: 16 / 9,
+  src: {
+    fullscreen: true,
+    autoSize: true,
+    setting: true,
+    whitelist: ["*"],
+  },
+};
+
+export default defineComponent({
+  name: "ArtPlayer",
+
+  props: {
+    /**
+     * artplayer src
+     */
+    src: {
+      type: Object as PropType<CustomArtPlayerOptions>,
+      default: optionsDefault.src,
+      required: false,
+    },
+
+    /**
+     * artplayer src.url
+     */
+    url: {
+      type: String,
+      default: "",
+      required: false,
+    },
+
+    /**
+     * artplayer src.poster
+     */
+    poster: {
+      type: String,
+      default: "",
+      required: false,
+    },
+
+    /**
+     * artplayer src.title
+     */
+    title: {
+      type: String,
+      default: "",
+      required: false,
+    },
+
+    /**
+     * artplayer src.muted
+     */
+    muted: {
+      type: Boolean,
+      default: false,
+      required: false,
+    },
+
+    /**
+     *  return artplayer
+     */
+    player: {
+      type: Function,
+      default: undefined,
+      required: false,
+    },
+
+    /**
+     * artplayer danmuku
+     */
+    pluginDanmuKu: {
+      type: Object,
+      default: () => ART_PLAYER.pluginDanmuKu || {},
+      required: false,
+    },
+
+    /**
+     * Component width
+     *
+     * 组件宽度
+     */
+    width: {
+      type: [String, Number],
+      default: ART_PLAYER?.width || optionsDefault.width || undefined,
+      required: false,
+    },
+
+    /**
+     * Component height
+     *
+     * 组件高度
+     */
+    height: {
+      type: [String, Number],
+      default: ART_PLAYER?.height || undefined,
+      required: false,
+    },
+
+    /**
+     * Component width / height ratio
+     *
+     * 组件长宽比
+     */
+    ratio: {
+      type: [String, Number],
+      default: ART_PLAYER?.ratio || optionsDefault.ratio || undefined,
+      required: false,
+    },
+  },
+
+  setup(props) {
+    const option = {
+      ...deepMerge(ART_PLAYER, props),
+    };
+
+    option.src ??= {};
+
+    const src = {
+      ...deepMerge(optionsDefault.src, option.src),
+    };
+
+    src.url = option.url ?? src.url ?? "";
+    src.poster = option.poster ?? src.poster ?? "";
+    src.title = option.title ?? src.title ?? "";
+    src.muted = option.muted ?? src.muted ?? false;
+
+    const { el, width, height } = useSize<HTMLDivElement>(props, 0);
+
+    let artplayer: ArtPlayer;
+
+    const initArtPlayer = async (): Promise<void> => {
+      const { default: art } = await import("artplayer/dist/artplayer.js");
+
+      src.container = el.value!;
+      src.customType ??= {};
+      src.type ??= autoType(src.url!) || "";
+
+      if (src.type) {
+        switch (src.type) {
+          case "m3u8":
+          case "hls":
+            src.customType[src.type] = m3u8;
+            break;
+          case "flv":
+            src.customType[src.type] = flv;
```

Comment:
I think I see similar code in utils?

Reply:
> I think I see similar code in utils?

In utils, the type is analyzed through the url. If the type is not set, the corresponding type will be added according to the url.
But here is to set the corresponding mse support by type, each player will have different codes